### DISCUSSION TITLE
How is --preferred-number-of-modules implemented?

### DISCUSSION THREAD

**Discussion by hartonen** [mapequation/infomap#144] [Q&A] [answered]
Hi!

I have been trying to look for how the penalty term penalizing for solutions with more (or less) modules than expected is implemented (the flag --preferred-number-of-modules), but was not able to find it from the publications. Could you kindly help me find the correct reference? Basically I am interested in seeing the Map Equation including the penalty term.

Thank you for your time and for the exceptionally well implemented and documented web page mapequation.org !

Best,
Tuomo

**Comment by danieledler** [comment, marked as answer]
Hi!

Thanks for your nice words about mapequation.org! Regarding the `--preferred-number-of-modules` there is not yet any paper exploring that or any other documentation, but the current implementation is simply adding `delta` bits extra as penalty, where `delta = abs(M_current - M_preferred)`.

If you have any other thoughts on this, please share!

Thanks!

**Comment by hartonen** [comment]
Thank you for a quick reply!

Could you specify what M_current and M_preferred stand for? Are they just the number of modules, i.e. M_current = number of modules in current partition and M_preferred the preferred number? Because in the papers symbol M seems to be reserved for a partition (L(M) being the description length of partition M) and it is not clear to me how in that case two partitions would be added together.

I don't have any concrete thoughts about it at the moment, but I will share should anything concrete come up.

**Comment by danieledler** [comment]
Ah, yes sorry I forgot to specify that, you are correct that they simply refer to the current or preferred number of modules.

Thanks!
